tutorials/other/ssh_keys.md

```diff
@@ -0,0 +1,91 @@
+Copyright 2015 [Nicky Semenza](http://nickysemenza.com)
+
+#SSH Keys
+
+SSH keys are an easy and secure way of authenticating with a server, without the use of a username or password
+
+
+There are two great uses for them (among others):
+
+ * Authenticating with github (password-less pushing)
+ * Authenticating with servers
+ 
+ 
+##How they work
+
+Basically, on the machine you want to connect to other servers with, you need to generate a public key file and a private key file.
+You then give your public key (id_rsa.pub) to the server you want to connect to, and it adds the key to its 'whitelist', known as the authorized_keys file ***never*** give your private key to anyone, treat it like you would with a secure password.
+##Generating Keys
+### (OSX & Linux based Systems)
+open up terminal, and run the command `ssh-keygen`
+It will ask you where you want to save it, and for an optional passphrase & confirmation of passphrase.
+
+You can just press enter 3 times to accept the defaults.
+
+The output should be similar to this:
+
+	Generating public/private rsa key pair.
+	Enter file in which to save the key (/home/nickysemenza/.ssh/id_rsa):
+	Created directory '/home/nickysemenza/.ssh'.
+	Enter passphrase (empty for no passphrase):
+	Enter same passphrase again:
+	Your identification has been saved in /home/nickysemenza/.ssh/id_rsa.
+	Your public key has been saved in /home/nickysemenza/.ssh/id_rsa.pub.
+	The key fingerprint is:
+	f4:55:64:37:72:23:96:28:9c:58:9f:72:7e:39:b6:2b nickysemenza@ubuntu
+	The key's randomart image is:
+	+--[ RSA 2048]----+
+	|        +.. .==+.|
+	|       . +..o++.o|
+	|        ...+.    |
+	|       . .+.  .  |
+	|        S .. =   |
+	|            o o  |
+	|             .   |
+	|           E  .  |
+	|            ..   |
+	+-----------------+
+	+-----------------+
+
+Now, if you cd into `~/.ssh` and list the files (`ls`), you should see at least 2 files
+
+	~/.ssh » ls                                                                                                                                                                                           
+	id_rsa  id_rsa.pub
+
+`id_rsa` is your private key file, and `id_rsa.pub` is your public key. Remember not to distribute the private key!
+
+###Windows
+Go download [both PuTTY and PuTTYGen](http://www.chiark.greenend.org.uk/~sgtatham/putty/)
+open PuTTYGen, click Generate, and then save the public and private keys somewhere of your choosing
+
+##Remote Machine Setup
+
+###OSX/Linux
+In order to make it so that you don't need to type your password when you SSH into the CS machines, we need to add your public key to the server's authorized_keys file.
+
+In one command, you can append your public key to the authorized_keys file on data.cs.purdue.edu
+
+`cat ~/.ssh/id_rsa.pub | ssh <yourusername>@data.cs.purdue.edu "cat >> ~/.ssh/authorized_keys"`
```

Comment:
ssh-copy-id isn't a preinstalled program on OSX though. We could suggest that people go download a [port of it?](https://github.com/beautifulcode/ssh-copy-id-for-OSX)